accounting/gst.js

```diff
@@ -0,0 +1,197 @@
+import frappe from 'frappejs';
+import { _ } from 'frappejs/utils';
+import { IPC_ACTIONS } from '@/messages';
+import { ipcRenderer } from 'electron';
+import { DateTime } from 'luxon';
+import { sleep } from 'frappejs/utils';
+import { makeJSON, showMessageDialog } from '@/utils';
+
+/**
+ * GST is a map which gives a final rate for any given gst item
+ * eg: IGST-18 = 18
+ * eg: GST-18 = CGST-9 + SGST-9 = 18
+ */
+const GST = {
+  'GST-0': 0,
+  'GST-0.25': 0.25,
+  'GST-3': 3,
+  'GST-5': 5,
+  'GST-6': 6,
+  'GST-12': 12,
+  'GST-18': 18,
+  'GST-28': 28,
+  'IGST-0': 0,
+  'IGST-0.25': 0.25,
+  'IGST-3': 3,
+  'IGST-5': 5,
+  'IGST-6': 6,
+  'IGST-12': 12,
+  'IGST-18': 18,
+  'IGST-28': 28,
+};
+
+/**
+ * CSGST is a map which return the tax rate component for state or central
+ * eg: GST-12 = 6
+ */
+const CSGST = {
+  'GST-0': 0,
+  'GST-0.25': 0.125,
+  'GST-3': 1.5,
+  'GST-5': 2.5,
+  'GST-6': 3,
+  'GST-12': 6,
+  'GST-18': 9,
+  'GST-28': 14,
+};
+
+/**
+ * IGST is a map which return the tax rate for the igst item
+ * eg: IGST-18 = 18
+ */
+const IGST = {
+  'IGST-0.25': 0.25,
+  'IGST-3': 3,
+  'IGST-5': 5,
+  'IGST-6': 6,
+  'IGST-12': 12,
+  'IGST-18': 18,
+  'IGST-28': 28,
+};
+
+export async function generateGstr1Json(report, { transferType, toDate }) {
+  const printSettings = await frappe.getSingle('PrintSettings');
+
+  if (!printSettings.gstin) {
+    promptWhenGstUnavailable();
+    return;
+  }
+
+  const savePath = await getSavePath();
```

Comment:
send the name from here 
```suggestion
  const savePath = await getSavePath('gstr-1');
```